### PR TITLE
MINOR: Modified snowflake view definition query

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes:

`SHOW VIEWS` does not incorporate the changes of a view in case it has been modified using `ALTER VIEW` statement.
We'll use the [GET_DDL](https://docs.snowflake.com/en/sql-reference/functions/get_ddl) method to get the correct view definitions.

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/snowflake/utils.py [hunk 1]
```diff
@@ -121,13 +121,10 @@ def get_view_definition(  # pylint: disable=unused-argument
     schema = schema or self.default_schema_name
     if schema:
         cursor = connection.execute(
-            "SHOW /* sqlalchemy:get_view_definition */ VIEWS "
-            f"LIKE '{view_name}' IN {schema}"
+            f"SELECT GET_DDL('VIEW','{schema}.{view_name}') AS \"text\""
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/database/snowflake/utils.py:124]: should we have something like this

/* {{"app": "OpenMetadata", %%}} */

Comment (OnkarVO7) [reply ingestion/src/metadata/ingestion/source/database/snowflake/utils.py]: Added it

File: ingestion/src/metadata/ingestion/source/database/snowflake/utils.py [hunk 2]
```diff
@@ -121,12 +121,11 @@ def get_view_definition(  # pylint: disable=unused-argument
...
     if schema:
         cursor = connection.execute(
-            "SHOW /* sqlalchemy:get_view_definition */ VIEWS "
-            f"LIKE '{view_name}' IN {schema}"
+            f'SELECT /* "app": "OpenMetadata" */ GET_DDL(\'VIEW\',\'{schema}.{view_name}\') AS "text"'
         )
     else:
         cursor = connection.execute(
-            "SHOW /* sqlalchemy:get_view_definition */ VIEWS " f"LIKE '{view_name}'"
+            f'SELECT /* "app": "OpenMetadata" */ GET_DDL(\'VIEW\',\'{view_name}\') AS "text"'
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/snowflake/utils.py:128]: aren't we including the comments directly on the engine?

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/database/snowflake/utils.py:128]: I think that if we use the connection engine, it'll be handled by this https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/connections/headers.py#L35 so no need to add it explicitely. Note how we don't add it anywhere

Comment (OnkarVO7) [reply ingestion/src/metadata/ingestion/source/database/snowflake/utils.py:128]: Ah makes sense thanks for reviewing it. Let me revert it

**Review by ayush-shah — APPROVED** [review]

**Pull request merged**